### PR TITLE
How to build a client server application using Rust

### LINKED ISSUE #5334: How to build a client server application using Rust
#### Proposed title of article

How to build a client-server application using Rust 

#### Proposed article introduction

A web app is a service built on top of the Internet to allow computers to share and exchange data easily and reliably. In this case, the web services will be referred to as a client-server model. Communications are established between the two to exchange data. Clients will ask for data, and the computer that serves this data is called a server. This blog will discuss the concept of the client-server model as we implement the whole scenario using Rust.

The reader will be able to understand the concept of the client-server communication. The reader will then learn how to achieve this using Rust.

#### Key takeaways 

- What is a Client
- What is a Server
- How Client-Server works
- Set up Rust
- Create a Client with Rust
- Create a Server with Rust
- Exchange data between the Client and Server

#### Article quality

The reader will be able to understand the concept of the client-server communication. The reader will then learn how to achieve this using Rust.

### PR DESCRIPTION
**Pull request by wanjcarol**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #5334

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by EKahuha** [comment]
I would like to review this @WanjaMIKE

**Comment by lalith1403** [comment]
plagiarism check done ✔️

File: content/articles/how-to-build-a-client-server-application-using-rust/index.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+This command will compile, run the application for you and print the result on your terminal.
+
+![cargo-run](/engineering-education/how-to-build-a-client-server-application-using-rust/cargo-run.png)
+
+If you have a problem setting up Rust, check [this tutorial](https://www.youtube.com/watch?v=enk0o7eWNsc) and learn how to install the rust utilities and tools to compile and create your rust programs.
+
+### The Rust modules
+
+We need a couple of Rust modules to set up this server and client. This includes;
```
Comment (EKahuha) [inline content/articles/how-to-build-a-client-server-application-using-rust/index.md:58]: These include;

File: content/articles/how-to-build-a-client-server-application-using-rust/index.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+use std::io;
+use std::time;
+use std ::net::{TcpListener,TcpStream};
+use std::io::{Read,Write};
+use std::thread;
+```
+
+#### Step Two - Handle the sender message
+
+Before setting up a server, the server needs to handle the client's messages. The server has first to access these messages then decide what operation to perform.
```
Comment (EKahuha) [inline content/articles/how-to-build-a-client-server-application-using-rust/index.md:83]: The server must first access these messages ....

**Review by EKahuha — CHANGES_REQUESTED** [review]
Great article @wanjcarol 
Make the recommended changes so that we can move on to tech review.

**Comment by wanjcarol** [comment]
@EKahuha, I have made the above changes.

**Comment by EKahuha** [comment]
This tutorial is ready for final review. @WanjaMIKE

**Review by EKahuha — APPROVED** [review]

**Comment by hectorkambow** [comment]
I'll leave the Final Review to the CM team - as they will assess the test reviews. 
@section-engineering-education/content-moderator-team  

cc @ahmadmardeni1 @Kayere @LinusMuema  

🚀

**Comment by ahmadmardeni1** [comment]
I'm on it @hectorkambow 🚀

**Comment by ahmadmardeni1** [comment]
Great tutorial. Please go through all the changes I've made. @wanjcarol 
cc @EKahuha Do the same. Please make sure you go through our [Peer Reviewers](https://github.com/section-engineering-education/engineering-education/tree/master/peer_reviewers) docs before your next test review.

**Pull request merged**